src/components/graph/graph.helper.js

```diff
@@ -323,17 +326,26 @@ function checkForGraphConfigChanges(nextProps, currentState) {
  * @returns {string|undefined} transform rule to apply.
  * @memberof Graph/helper
  */
-function getCenterAndZoomTransformation(d3Node, config) {
+function getCenterAndZoomTransformation(d3Node, config, containerElId) {
     if (!d3Node) {
         return;
     }
 
     const { width, height, focusZoom } = config;
 
+    const transform = {
+        x: width / 2 - d3Node.x,
+        y: height / 2 - d3Node.y,
+        k: focusZoom,
+    };
+
+    const selector = d3Select(`#${containerElId}`);
+    // in order to initilize the new position
+    selector.call(d3Zoom().transform, d3.zoomIdentity.translate(transform.x, transform.y).scale(transform.k));
```

Comment:
I'll need to verify that focusAnimationDuration doesn't break with this change. @LonelyPrincess if you have some time could you please have a look at this one? I remember you implemented this one initially. Thanks!